otel/internal/utils/sentryrequest.go

```diff
@@ -27,7 +27,10 @@ func IsSentryRequestSpan(ctx context.Context, s trace.ReadOnlySpan) bool {
 func isSentryRequestUrl(ctx context.Context, url string) bool {
 	hub := sentry.GetHubFromContext(ctx)
 	if hub == nil {
-		return false
+		hub = sentry.CurrentHub()
```

Comment:
Note: for better experience and to fix more cases like this, we might want to set the Hub on context a lot earlier, maybe even in StartSpan.

Reply:
We might not know which hub to set on the `ctx` in `StartSpan`.
I think it might be worth digging a little bit into how Otel handles go routines.
For now, we might still consider this as a quick win.